src/charm.py

```diff
@@ -340,12 +332,18 @@ def _update_workload(self, event: HookEvent) -> None:
             return
 
         self._create_token()
+
+        if not self._container.isdir(LOG_FOLDER):
+            self._container.make_dir(path=LOG_FOLDER, make_parents=True)
+            logger.info(f"Created directory {LOG_FOLDER}")
+
         if not self.model.relations[DATABASE_RELATION_NAME]:
             self.unit.status = BlockedStatus("Missing required relation with postgresql")
             return
 
         if not self._dsn:
             self.unit.status = WaitingStatus("Waiting for database creation")
+            event.defer()
```

Comment:
how would it work if we had to drop `defer`?

Reply:
Tbh I think that it should be fixed event without the defer. The problem mainly was the logic in the `database_created` method (the decorator). I can try to remove and see how it goes, as an exercise.